### PR TITLE
ci: extend commitlint in cli

### PR DESCRIPTION
**Pull request by huxuan**
By configuring the commitlint in cli directly, we do not need to configure it in `package.json` and even do not need to checkout the repo.

In the meanwhile, we can rename the `package.json` to `.releaserc.json` as more explicitly configuration for semantic-release.

BTW, since I have no experience about js code style or format, I would prefer to use `json` configuration so as to leverage those json lint tools.

### CODE REVIEW

File: .github/workflows/commitlint.yml [hunk 1]
```diff
@@ -9,13 +9,12 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - run: env | sort
-      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
       - name: Validate the latest commit message with commitlint
         if: github.event_name == 'push'
-        run: echo "${{ github.event.head_commit.message }}" | npx commitlint
+        run: echo "${{ github.event.head_commit.message }}" | npx commitlint -x @commitlint/config-conventional
```
Comment (msclock) [inline .github/workflows/commitlint.yml:14]: why not use npx --package same as the semantic-release.🤔

**Comment by msclock** [comment]
>  rename the package.json to .releaserc.json as more explicitly configuration for semantic-release.

we are back here again😅

File: .github/workflows/commitlint.yml (hunk 1 above)
Comment (huxuan) [reply .github/workflows/commitlint.yml:14]: Please refer to this: https://github.com/serious-scaffold/ss-python/pull/562

Now we are useing the container instead, so it is also managed by renovate, too.

**Comment by huxuan** [comment]
> > rename the package.json to .releaserc.json as more explicitly configuration for semantic-release.
> 
> we are back here again😅

Yes, you are right. I usually prefer universal configuration if there are multiple tools needed. Since we do not need the configure for commitlint now, it is better to make it more explicitly.

File: .github/workflows/commitlint.yml (hunk 1 above)
Comment (msclock) [reply .github/workflows/commitlint.yml:14]: Got it.😬

**Review by msclock — APPROVED** [review]

**Pull request merged**